src/mon/OSDMonitor.cc

```diff
@@ -12648,22 +12668,30 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       goto reply;
     }
 
-    if (expected_num_objects > 0 &&
-	cct->_conf->osd_objectstore == "filestore" &&
-	cct->_conf->filestore_merge_threshold > 0) {
+    bool osd_all_bluestore = is_osd_all_bluestore();
+    if (!osd_all_bluestore &&
+        expected_num_objects > 0 &&
+        cct->_conf->osd_objectstore == "filestore" &&
```

Comment:
might want to drop this line. as it does not matter what `osd_objectstore` is on the monitor node.